src/main.ts

```diff
@@ -56,28 +56,30 @@ export async function activate(context: vscode.ExtensionContext) {
 		},
 	};
 
+	let packageInfo = Utils.getPackageInfo();
 	let commandObserver = new CommandObserver();
 	const installationStart = Date.now();
 	serverdownloader.getOrDownloadServer().then(e => {
-		vscode.window.showErrorMessage(e);
 		const installationComplete = Date.now();
 		let serverOptions = generateServerOptions(e);
-		languageClient = new SqlOpsDataClient(Constants.serviceName, serverOptions, clientOptions);
-		commandObserver._client = languageClient;
-		addDeployNotificationsHandler(commandObserver);
-		const processStart = Date.now();
-		languageClient.onReady().then(() => {
-			const processEnd = Date.now();
-			statusView.text = 'Pgsql service started';
-			setTimeout(() => {
-				statusView.hide();
-			}, 1500);
-			Telemetry.sendTelemetryEvent('startup/LanguageClientStarted', {
-				installationTime: String(installationComplete - installationStart),
-				processStartupTime: String(processEnd - processStart),
-				totalTime: String(processEnd - installationStart),
-				beginningTimestamp: String(installationStart)
-			});
+        languageClient = new SqlOpsDataClient(Constants.serviceName, serverOptions, clientOptions);
+        for (let command of registerCommands(commandObserver, packageInfo, languageClient)) {
```

Comment:
Is it intentional to move the registration to the success handler? Are there still any commands possible, if python back-end doesn't work? Alternative way could be to keep it where it was, but internally check if passed ```languageClient``` is not null and register additional commands that depend on it. But if there are no commands that can successfully execute, then it doesn't matter probably.